### ISSUE TITLE
One model works, the other doesnt. Both are virtually identical.

### ISSUE THREAD

**Issue by Shaunakde** [BVLC/caffe#2323]
I am trying to build a minimal example of a neural network with HDF5 data that I have prepared from a CSV file using the caffe libraries.

My solver is as follows:

    net: "examples/wine/wine_train.prototxt"
    test_iter: 250
    test_interval: 1000
    base_lr: 0.01
    lr_policy: "step"
    gamma: 0.1
    stepsize: 5000
    display: 1000
    max_iter: 10000
    momentum: 0.9
    weight_decay: 0.0005
    snapshot: 10000
    snapshot_prefix: "examples/wine/train_"
    solver_mode: CPU

This model runs fine:

    name: "WineNet"
    layer {
      name: "data"
      type: "HDF5Data"
      top: "wine_train_data"
      top: "wine_train_label"
      hdf5_data_param {
        source: "examples/wine/train.txt"
        batch_size: 10
      }
    }
    layer {
      name: "fc1"
      type: "InnerProduct"
      bottom: "wine_train_data"
      top: "fc1"
      inner_product_param {
        num_output: 2
    	}
    }
    layer {
      name: "loss"
      type: "SoftmaxWithLoss"
      bottom: "fc1"
      bottom: "wine_train_label"
      top: "loss"
    }

This one gives an error:

    name:"wineclass"
    layers {
      name: "data"
      type: "HDF5Data"
      top: "wine_train_data"
      top: "wine_train_label"
      hdf5_data_param {
        source: "examples/wine/train.txt"
        batch_size: 10
      }
    }
    layer {
      name: "ip"
      type: "InnerProduct"
      bottom: "wine_train_data"
      top: "ip"
      inner_product_param {
        num_output: 2
      }
    }
    layer {
      name: "loss"
      type: "SoftmaxWithLoss"
      bottom: "ip"
      bottom: "wine_train_label"
      top: "loss"
    }

Each time I get the following error:

    shaunak@ubuntu:~/caffe$ build/tools/caffe train -model '/home/shaunak/caffe/examples/wine/wine_train.prototxt' -solver '/home/shaunak/caffe/examples/wine/solver.prototxt'
    I0415 04:31:00.154145 57047 caffe.cpp:117] Use CPU.
    I0415 04:31:00.154485 57047 caffe.cpp:121] Starting Optimization
    I0415 04:31:00.154552 57047 solver.cpp:32] Initializing solver from parameters:
    test_iter: 250
    test_interval: 1000
    base_lr: 0.01
    display: 1000
    max_iter: 10000
    lr_policy: "step"
    gamma: 0.1
    momentum: 0.9
    weight_decay: 0.0005
    stepsize: 5000
    snapshot: 10000
    snapshot_prefix: "wine/train"
    solver_mode: CPU
    net: "examples/wine/wine_train.prototxt"
    I0415 04:31:00.154660 57047 solver.cpp:70] Creating training net from net file: examples/wine/wine_train.prototxt
    [libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 4:9: Expected integer or identifier.
    F0415 04:31:00.154774 57047 upgrade_proto.cpp:928] Check failed: ReadProtoFromTextFile(param_file, param) Failed to parse NetParameter file: examples/wine/wine_train.prototxt
    *** Check failure stack trace: ***
        @     0x7f4a30766c3c  google::LogMessage::Fail()
        @     0x7f4a30766b88  google::LogMessage::SendToLog()
        @     0x7f4a3076658a  google::LogMessage::Flush()
        @     0x7f4a30769521  google::LogMessageFatal::~LogMessageFatal()
        @     0x7f4a30b8b1ee  caffe::ReadNetParamsFromTextFileOrDie()
        @     0x7f4a30b6dfa2  caffe::Solver<>::InitTrainNet()
        @     0x7f4a30b6ee63  caffe::Solver<>::Init()
        @     0x7f4a30b6f036  caffe::Solver<>::Solver()
        @           0x40c3c0  caffe::GetSolver<>()
        @           0x406361  train()
        @           0x4048f1  main
        @     0x7f4a2fe86ec5  (unknown)
        @           0x404e9d  (unknown)
    Aborted (core dumped)
    shaunak@ubuntu:~/caffe$

Virtually identical models: One works the other doesn't. I cant explain why!

First posted to SO: http://stackoverflow.com/questions/29698136/caffe-hdf5-neural-network-basic-example-fails-to-parse-the-model-file

**Comment by userfou** [comment]
in your solver you are pointing at the model "examples/wine/wine_train.prototxt". Maybe the first model is called so, but the second one is called something else or is not in the same directory ?

**Comment by Shaunakde** [comment]
No. I'm swapping both in the same text file so the com

**Comment by userfou** [comment]
Oh ok. Than I think I can not help ^^. Maybe try to change the words "layers" to "layer" without s in your second model (I don't think it's not the problem, but it's the only difference I see between both models)

Good Luck

**Comment by Shaunakde** [comment]
No. I'm swapping both in the same text file so the command remains exactly
the same.

**Comment by Shaunakde** [comment]
Apologies. That was a typographical error in the question surprisingly. ( Stack overflow markdown )

I have added the file I am using in my previous reply. 

Thanks for looking into this.

**Comment by Shaunakde** [comment]
@userfou layers was the problem. Was overlooking the first one.
